code/modules/chemistry/Chemistry-Tools.dm

```diff
@@ -996,3 +996,122 @@ proc/ui_describe_reagents(atom/A)
 		amount_of_reagent_to_use = 5 //slow but...
 		chemical_efficiency = 2 //...lots of synthflesh per unit of blood
 		organ_efficiency = 2
+
+/obj/item/bunsen_burner
+	name = "Bunsen burner"
+	desc = "A Bunsen burner capable of holding up chemical containers and heating them at three different heat levels."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bunsen"
+	w_class = W_CLASS_NORMAL
+	var/temperature_setting = "low" //medium, high
+	var/is_currently_burning = FALSE //keep seperate from temp setting so it stays on the same setting if you turn it off and on
+	var/obj/item/reagent_containers/current_container = null
+	var/list/datum/contextAction/contexts = list()
+	var/image/light_image = null
+	var/current_temp_to_heat = 20 //!how much the bunsen burner heats per processing tick
+	var/current_temp_cap = 110 //!how hot the container will have to be before it stops heating
+	contextLayout = new /datum/contextLayout/experimentalcircle
+
+	New()
+		contexts += new /datum/contextAction/bunsen/heat_low
+		contexts += new /datum/contextAction/bunsen/heat_medium
+		contexts += new /datum/contextAction/bunsen/heat_high
+		contexts += new /datum/contextAction/bunsen/heat_off
+		..()
+
+	disposing()
+		STOP_TRACKING
+		..()
+
+	process()
+		if(is_currently_burning && current_container)
+			heat_container()
+		..()
+
+	update_icon()
+		if(!is_currently_burning)
+			icon_state = "bunsen"
+		else switch(temperature_setting)
+			if("low")
+				icon_state = "bunsen_low"
+			if("medium")
+				icon_state = "bunsen_medium"
+			if("high")
+				icon_state = "bunsen_high"
+		if(current_container)
+			if (!src.light_image)
+				src.light_image = image(src.icon)
+				src.light_image.appearance_flags = PIXEL_SCALE | RESET_COLOR | RESET_ALPHA
+				src.light_image.icon_state = "bunsen_light"
+			src.UpdateOverlays(light_image, "light")
+		else
+			src.UpdateOverlays(null, "light")
+
+	proc/change_status(var/status)
+		src.remove_simple_light("bunsen_flame")
+		if(status != "off")
+			switch(status)
+				if("low")
+					add_simple_light("bunsen_flame", list(245, 181, 61, 0.7 * 255))
```

Comment:
if you change this to use `/datum/component/loctargeting/simple_light` it'll work for people holding it too